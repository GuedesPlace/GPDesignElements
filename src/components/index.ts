export * from './navigation';
export * from './page';
export * from './ui-elements';
export * from './pictureuploader';