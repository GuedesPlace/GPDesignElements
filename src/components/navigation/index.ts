export * from './GPLayout';
export * from './HeaderLine';