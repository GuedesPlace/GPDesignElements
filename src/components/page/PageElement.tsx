import { INavLink, INavLinkGroup, INavStyles, Nav, Stack, StackItem, Text } from "@fluentui/react";
import React, { PropsWithChildren } from "react";

export interface IPageProps {
    title:string;
    navigationItems?:INavLinkGroup[];
    navigationStyles?:Partial<INavStyles>;
    onNavigationItemClick?:(ev?: React.MouseEvent<HTMLElement>, item?: INavLink)=>void
}

export const Page:React.FunctionComponent<PropsWithChildren<IPageProps>> = (props:PropsWithChildren<IPageProps>) => {
    return <Stack style={{margin:"15px"}} horizontal>
        {props.navigationItems ? 
            <StackItem><Nav 
                groups={props.navigationItems} 
                styles={props.navigationStyles} 
                onLinkClick={props.onNavigationItemClick}/>
            </StackItem>
            :null}
        <StackItem grow={10}>
            <Stack tokens={{childrenGap:"5px"}}>
                <StackItem><Text variant="xLarge">{props.title}</Text></StackItem>
                <StackItem><hr></hr></StackItem>
                <StackItem>{props.children}</StackItem>
            </Stack>
        </StackItem>
    </Stack>
}