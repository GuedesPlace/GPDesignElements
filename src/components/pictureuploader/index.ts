export * from './PictureUploaderElement';
export * from './DropZone';