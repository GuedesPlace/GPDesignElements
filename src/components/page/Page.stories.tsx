import React from "react";
import { Meta, StoryObj } from "@storybook/react";
import { Page } from "./PageElement";
import { Text } from "@fluentui/react";

const meta: Meta<typeof Page> = {
    component: Page,
    title: "Basic/Page",
    argTypes: {},
};
export default meta;

type Story = StoryObj<typeof Page>;

export const Basic: Story = (args: any) => (
    <Page data-testId="InputField-id" {...args} />
);
Basic.args = {
    title: "Page minimal Attributes setted"
};

export const BasicWithLoremIpsumText: Story = (args: any) => (
    <Page data-testId="InputField-id" {...args} />
);
BasicWithLoremIpsumText.args = {
    title: "Page minimal Attributes setted",
    children: <Text variant="medium">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </Text>
};

export const BasicWithNavigationAndLoremIpsumText: Story = (args: any) => (
    <Page data-testId="InputField-id" {...args} />
);
BasicWithNavigationAndLoremIpsumText.args = {
    title: "Page minimal Attributes setted",
    children: <Text variant="medium">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </Text>,
    navigationItems:[
        {
          links: [
            {
              name: 'Home',
              url: 'http://example.com',
              expandAriaLabel: 'Expand Home section',
              links: [
                {
                  name: 'Activity',
                  url: 'http://msn.com',
                  key: 'key1',
                  target: '_blank',
                },
                {
                  name: 'MSN',
                  url: 'http://msn.com',
                  disabled: true,
                  key: 'key2',
                  target: '_blank',
                },
              ],
              isExpanded: true,
            },
            {
              name: 'Documents',
              url: 'http://example.com',
              key: 'key3',
              isExpanded: true,
              target: '_blank',
            },
            {
              name: 'Pages',
              url: 'http://msn.com',
              key: 'key4',
              target: '_blank',
            },
            {
              name: 'Notebook',
              url: 'http://msn.com',
              key: 'key5',
              disabled: true,
            },
            {
              name: 'Communication and Media',
              url: 'http://msn.com',
              key: 'key6',
              target: '_blank',
            },
            {
              name: 'News',
              url: 'http://cnn.com',
              icon: 'News',
              key: 'key7',
              target: '_blank',
            },
          ],
        },
      ]
};
