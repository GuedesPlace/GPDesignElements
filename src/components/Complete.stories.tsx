import React from "react";
import { Meta, StoryObj } from "@storybook/react";
import { Text } from "@fluentui/react";
import { Page } from "./page";
import { GPLayout } from "./navigation";

const meta: Meta<typeof GPLayout> = {
  component: GPLayout,
  title: "01. Complete/All",
  argTypes: {},
};
export default meta;

type Story = StoryObj<typeof GPLayout>;

export const ApplicationAndContent: Story = (args: any) => (
  <GPLayout data-testId="InputField-id" {...args} />
);
ApplicationAndContent.args = {
  title: "This is Loggedin",
  linkItems: [{
    id: 'dasboardId',
    title: 'The Dasboard',
    onClick: (id) => console.log(id)
  },
  {
    id: 'createId',
    title: 'CREATE',
    onClick: (id) => console.log(id)
  }
  ],
  userInformation: {
    commonName: "John Doe",
    eMail: "j@doe.family",
    oganisationName: "Does graeat big Family"
  },
  applicationItems: [
    { title: 'Photos', iconName: "PhotoVideoMedia", id: 'photos', onClick: () => console.log("photos") },
    { title: 'Profile', iconName: "ProfileSearch", id: 'profiles', onClick: (id) => console.log(id) }
  ],
  version: "0.1.9-beta22",
  applicationInformations: [{
    id: "memory",
    title: "Used Memory: 50 GB",
    onClick: (id) => console.log(id)
  }, {
    id: "memory2",
    title: "Free Memory: 5 GB",
    onClick: (id) => console.log(id)
  }],
  selectedApplicationId: "photos",
  children: <>
    <Page title="Page minimal Attributes setted"
      navigationItems={[
        {
          links: [
            {
              name: 'Home',
              url: 'http://example.com',
              expandAriaLabel: 'Expand Home section',
              links: [
                {
                  name: 'Activity',
                  url: 'http://msn.com',
                  key: 'key1',
                  target: '_blank',
                },
                {
                  name: 'MSN',
                  url: 'http://msn.com',
                  disabled: true,
                  key: 'key2',
                  target: '_blank',
                },
              ],
              isExpanded: true,
            },
            {
              name: 'Documents',
              url: 'http://example.com',
              key: 'key3',
              isExpanded: true,
              target: '_blank',
            },
            {
              name: 'Pages',
              url: 'http://msn.com',
              key: 'key4',
              target: '_blank',
            },
            {
              name: 'Notebook',
              url: 'http://msn.com',
              key: 'key5',
              disabled: true,
            },
            {
              name: 'Communication and Media',
              url: 'http://msn.com',
              key: 'key6',
              target: '_blank',
            },
            {
              name: 'News',
              url: 'http://cnn.com',
              icon: 'News',
              key: 'key7',
              target: '_blank',
            },
          ],
        },
      ]}><Text variant="medium">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </Text></Page></>
};


