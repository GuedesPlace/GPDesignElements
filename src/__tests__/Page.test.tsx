import renderer from 'react-test-renderer';
import {Page} from '../components/page';
import React from 'react';

it('changes the class when hovered', () => {
  const component = renderer.create(
   <Page title="Hello World"></Page>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});