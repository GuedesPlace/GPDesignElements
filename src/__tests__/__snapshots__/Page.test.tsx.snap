// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="ms-Stack css-109"
  style={
    Object {
      "margin": "15px",
    }
  }
>
  <div
    className="ms-StackItem css-110"
  >
    <div
      className="ms-Stack css-111"
    >
      <div
        className="ms-StackItem css-112"
      >
        <span
          className="css-113"
        >
          Hello World
        </span>
      </div>
      <div
        className="ms-StackItem css-112"
      >
        <hr />
      </div>
    </div>
  </div>
</div>
`;
